modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/redirect/button/RedirectButtonDDMFormFieldTemplateContextContributor.java

```diff
@@ -52,14 +52,17 @@
 
 		return HashMapBuilder.<String, Object>put(
 			"buttonLabel",
-			DDMFormFieldTypeUtil.getPropertyValue(
+			DDMFormFieldTypeUtil.getPropertyValues(
 				ddmFormField, ddmFormFieldRenderingContext.getLocale(),
-				"buttonLabel")
+				"buttonLabel")[0]
 		).put(
 			"message",
-			DDMFormFieldTypeUtil.getPropertyValue(
-				ddmFormField, ddmFormFieldRenderingContext.getLocale(),
-				"message")
+			LanguageUtil.format(
+				ResourceBundleUtil.getBundle(
+					ddmFormFieldRenderingContext.getLocale(), getClass()),
+				GetterUtil.getString(
+					((Object[])ddmFormField.getProperty("message"))[0]),
+				(Object[])ddmFormField.getProperty("messageArguments"))
```

Comment:
@rodrigopaulino, from what I could see, you've decided to create the class `DDMFormFieldProperty` to allow the property `messageArguments` support two arguments instead of just one, is my understanding correct?

If so, do you think we can solve the request demanded by Brian using the following examples?
- https://github.com/liferay/liferay-portal/blob/master/modules/apps/archived/journal-content-search-web/src/main/java/com/liferay/journal/content/search/web/internal/display/context/JournalContentSearchDisplayContext.java#L139-L143
- https://github.com/liferay/liferay-portal/blob/master/modules/apps/adaptive-media/adaptive-media-web/src/main/java/com/liferay/adaptive/media/web/internal/portlet/action/EditImageConfigurationEntryMVCActionCommand.java#L169-L177

I've discussed about this new proposed class`DDMFormFieldProperty` with Carol and we've both understood that the solutions linked above can be a less costly solution for Brian's request. Can you please analyze them?

Reply:
@natocesarrego, do you mean using the `Language.format` directly in the properties field of the `DDMFormField` annotation? If so, costly on what aspect? Don't you agree that developers would have more power to pass an array of values instead of just one? The example you're giving is only one example of having that need, but if you check right below, the same goes to URL parameters. In this use of the redirect button. we only use one parameter, but developers could want to attach more than one to the final `redirectURL`. Let me know what you think.

Reply:
Hey, @natocesarrego. Did you have any time to check my answer here? Thx

Reply:
No, @rodrigopaulino. Actually, considering the [first request from Brian](https://github.com/brianchandotcom/liferay-portal/pull/103947#issuecomment-874262604), the idea proposed by Carol and I is instead of creating a new property named `messageArguments` to support more than one argument to format a language key properly, you can give a try to use the `LanguageUtil.format` in the `RedirectButtonDDMFormFieldTemplateContextContributor.java`, like below for example:
```
LanguageUtil.format(
    resourceBundle, "a-google-places-api-key-is-required-to-use-this-field-x-x",
    new String[] {
        "<strong>", "</strong>"
    });
```
Does this code above attend Brian's request?

Carol did something similar in the front-end, as you can see [here](https://github.com/liferay/liferay-portal/commit/d619bdf28c203e594af48dfaae58868aaec17308). :slightly_smiling_face:

Reply:
Hi, @natocesarrego. Thanks for detailing what you meant before.
I'm afraid your suggestion would cause the Redirect Button to lose its ability of customising the message according to the developer's needs. In other words, all redirect buttons created after this would have the same message: "A Google Places API key is required to use this field. Go to **Site Settings → Third Party Applications** to set up a valid key."